### PR TITLE
Fix All warnings in "Usability:Typography", "Usability:Icons", and "Usability" categories

### LINKED ISSUE #5194: All warnings in "Usability:Typography", "Usability:Icons", and "Usability" categories
## Description
We would like to fix all the remaining lint issues on Oppia Android in the "Usability:Typography", "Usability:Icons", and "Usability" categories.

## Steps To Reproduce
Run the following command from a terminal in a checked-out Oppia Android repo:

./gradlew :app:lint

## Expected Behavior
Running the command above should not show any errors in this category

### PR DESCRIPTION
**Pull request by Rd4dev**
## Explanation

Fixes #5194 [part of #5182]

Fixes all warning in "Usability" category caused by

- **Borderless Buttons :** Fixed by introducing `style="?android:attr/buttonBarButtonStyle"` to the borderless buttons
- **OK text capitalization :** Fixed by suppressing the lint warnings for `ButtonCase` and `Typos`
- **Ellipsis/Hyphen unicode replacement :** Fixed by introducing unicode representations (under review)
- **Image in density drawable folder :** Fixed by creating seperate density images (under review)
- **Text size too small :** Fixed by upscaling text size to `12sp`
- **To incorporate Autofill :** Fixed by incorporating `autofillHints` to the EditText

## ScreenShot

Before After Usability Lint Issues

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

**Comment by adhiamboperes** [comment]
@Rd4dev, it seems that all the concerns have been addressed in the issue thread. Please update the PR and reassign for review.

**Comment by Rd4dev** [comment]
@adhiamboperes, as suggested in the [#5169](https://github.com/oppia/oppia-android/issues/5194#issuecomment-1815570224), I had kept the unicode changes -> [[changes mentioned]](https://github.com/oppia/oppia-android/issues/5194#issuecomment-1767703119), updated the PR with the latest changes and merged them.

**Comment by Rd4dev** [comment]
Hi @adhiamboperes, the task has already been assigned to @BenHenning, and I'm unable to find an option to reassign it for review. Is there a specific process or action I should take to assign a reviewer again?
Web capture_22-11-2023_19111_github com

**Comment by adhiamboperes** [comment]
@Rd4dev, per [wiki](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section), you need to do a top level comment like: "@ReviewerName, PTAL".  In this case, reviewer is me.

This is for the Assignees' section

**Comment by Rd4dev** [comment]
> @Rd4dev, per [wiki](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section), you need to do a top level comment like: "@ReviewerName, PTAL". In this case, reviewer is me.
> 
> This is for the Assignees' section

@adhiamboperes Got it, thanks! Will make sure to refer the wiki for future queries.

**Comment by Rd4dev** [comment]
@adhiamboperes PTAL

**Review by adhiamboperes — APPROVED** [review]
Thanks @Rd4dev, this LGTM!

**Comment by BenHenning** [comment]
Thanks @Rd4dev! Sorry for the delay in my review.

Per https://github.com/oppia/oppia-android/issues/5194#issuecomment-1767379775 it seems that most (all?) of the new PNGs are tied to lesson data. We shouldn't bother updating these as they are going to be deleted in the future, anyway. Is it possible for us to silence the warnings for these images, instead?

/cc @adhiamboperes

**Comment by Rd4dev** [comment]
@BenHenning, that's completely alright! 

As these images are expected to be changed, one effective approach to address/silence these warnings is by **relocating the images to the `drawable-nodpi` folder**. This folder signifies that the images should not undergo automatic density scaling, and it allows for the suppression of associated lint warnings.

I've implemented these adjustments in the latest commit and would appreciate your feedback on whether this could be considered a valid solution. 

**UI Screenshots:**

HomeLessons

/cc @adhiamboperes 

Thank you.

**Comment by Rd4dev** [comment]
@BenHenning PTAL

File: app/src/main/res/layout-sw600dp-land/topic_lessons_story_summary.xml [hunk 1]
```diff
@@ -86,7 +86,7 @@
               android:fontFamily="sans-serif"
               android:text="@{viewModel.storyPercentage != 100? @string/status_in_progress : @string/status_completed}"
               android:textColor="@color/component_color_shared_primary_text_color"
-              android:textSize="10sp"
+              android:textSize="12sp"
```
Comment (BenHenning) [inline app/src/main/res/layout-sw600dp-land/topic_lessons_story_summary.xml:89]: This is an actual stylistic change--why is it needed? I'm wondering if the warning is correct for this one.

Could you also add before/after screenshots showing this text size adjustment? That's helpful for reviewers trying to assess a visual change.

File: app/src/main/res/layout/survey_free_form_layout.xml [hunk 2]
```diff
@@ -24,7 +24,8 @@
       app:layout_constraintEnd_toEndOf="parent"
       app:layout_constraintStart_toStartOf="parent"
       app:layout_constraintTop_toTopOf="parent"
-      app:textChangedListener="@{viewModel.answerTextWatcher}" />
+      app:textChangedListener="@{viewModel.answerTextWatcher}"
+      android:autofillHints="@string/survey_free_form_text_hint"/>
```
Comment (BenHenning) [inline app/src/main/res/layout/survey_free_form_layout.xml]: Nit: need space before ``/``.

Also, was this something missed in an earlier PR @adhiamboperes?

File: app/src/main/res/mipmap-anydpi-v26/launcher_icon.xml [hunk 3]
```diff
@@ -20,4 +20,21 @@
       </group>
     </vector>
   </foreground>
+  <monochrome>
```
Comment (BenHenning) [inline app/src/main/res/mipmap-anydpi-v26/launcher_icon.xml:23]: Interesting--why was this change needed? Also, do you have a screenshot of what this monochrome version of the launcher icon looks like?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @Rd4dev! Apologies for my delayed review here. The PR generally looks really good--just had a few comments. PTAL.

File: app/src/main/res/layout-sw600dp-land/topic_lessons_story_summary.xml (hunk 1 above)
Comment (Rd4dev) [reply app/src/main/res/layout-sw600dp-land/topic_lessons_story_summary.xml:89]: The warning mentions: **Text size too small**
Description: Avoid using sizes smaller than 11sp: 10sp

To address the warning, I opted for a font size exceeding 11sp. Alternatively, one can suppress the warnings. 
Reference images for both sizes are provided below.

Text Size: **10sp**
Text Size 10sp

Text Size: **12sp**
Text Size 12sp

**Before/After:**
Before After

File: app/src/main/res/layout/survey_free_form_layout.xml (hunk 2 above)
Comment (Rd4dev) [reply app/src/main/res/layout/survey_free_form_layout.xml]: Through the [commit](https://github.com/oppia/oppia-android/pull/5196/commits/04624c7d293ed1b2ed22f03256ddab5db2f5cddb) , Introduced a space ahead of the /> closing tag.

File: app/src/main/res/mipmap-anydpi-v26/launcher_icon.xml (hunk 3 above)
Comment (Rd4dev) [reply app/src/main/res/mipmap-anydpi-v26/launcher_icon.xml:23]: The lint warning indicates:  **Monochrome icon is not defined**. 

Android 13 has introduced the _Themed Icons_ feature, allowing the tinting of all supported app icons with the same color palette. This can be achieved through the _monochromatic version_ of the adaptive icon. Failure to include this may lead to the display of the same default icon across all themes. 

Reference icons have been provided below:
Before Text Size 10sp (1)

**Comment by Rd4dev** [comment]
@adhiamboperes PTAL - [reviews](https://github.com/oppia/oppia-android/pull/5196#pullrequestreview-1778398585)

**Comment by Rd4dev** [comment]
@BenHenning PTAL - [review](https://github.com/oppia/oppia-android/pull/5196#pullrequestreview-1778398585)

File: app/src/main/res/layout-sw600dp-land/topic_lessons_story_summary.xml (hunk 1 above)
Comment (adhiamboperes) [reply app/src/main/res/layout-sw600dp-land/topic_lessons_story_summary.xml:89]: @BenHenning, for this, I would go for changing the size to 12sp because it is android best practice, per lint warning.  I think this is also required for accessibility reasons.

File: app/src/main/res/mipmap-anydpi-v26/launcher_icon.xml (hunk 3 above)
Comment (BenHenning) [reply app/src/main/res/mipmap-anydpi-v26/launcher_icon.xml:23]: Ah got it. Thanks for the additional context.

File: app/src/main/res/layout-sw600dp-land/topic_lessons_story_summary.xml (hunk 1 above)
Comment (BenHenning) [reply app/src/main/res/layout-sw600dp-land/topic_lessons_story_summary.xml:89]: Yeah I agree. Plus it still looks good.

**Review by BenHenning — APPROVED** [review]
Thanks @Rd4dev! Really sorry about the delayed review. The PR LGTM.
